File: csharp/src/api/Options/RequestResponseLoggingOptions.cs
namespace Example.Api.Options;

/// <summary>
/// Options for request and response logging.
/// </summary>
public class RequestResponseLoggingOptions
{
    /// <summary>
    /// Indicates whether request logging is enabled.
    /// </summary>
    /// <value></value>
    public bool EnabledRequestLog { get; set; }

    /// <summary>
    /// Indicates whether response logging is enabled.
    /// </summary>
    /// <value></value>
    public bool EnabledResponseLog { get; set; }
}

File: src/nodejs/src/config/RequestResponseLoggingOptions.ts
/**
 * Options for request and response logging middleware.
 */
export interface IRequestResponseLoggingOptions {
  /**
   * Enable request body logging.
   */
  enabledRequestLog: boolean;

  /**
   * Enable response body logging.
   */
  enabledResponseLog: boolean;
}

/**
 * Default options for request and response logging.
 */
export const defaultRequestResponseLoggingOptions: IRequestResponseLoggingOptions = {
  enabledRequestLog: process.env.LOG_REQUEST_BODY === "true",
  enabledResponseLog: process.env.LOG_RESPONSE_BODY === "true",
};
